### PR TITLE
Display specific banner on downloads page if manifest source unavailable

### PR DESCRIPTION
**Pull request by lowellrex**
Connects #678. Show a more specific banner on the downloads progress page if a manifest source is unavailable.

image

### CODE REVIEW

File: client/src/Constants.js [hunk 1]
```diff
@@ -20,6 +20,6 @@ export const MANIFEST_DOWNLOAD_STATE = {
 export const MANIFEST_SOURCE_FETCH_STATE = {
   NOT_STARTED: 'initialized',
   IN_PROGRESS: 'pending',
-  SUCCEEDED: 'finished',
+  SUCCEEDED: 'success',
```
Comment (lowellrex) [inline client/src/Constants.js]: This was mapped incorrectly previously

**Comment by lowellrex** [comment]
I thought we would need to add a new icon for this change so I went ahead and refactored all of the different Icons into a single file. Even though we didn't end up needing to add a new icon I decided to leave the Icons refactor in this PR because I think it is a good refactor. Happy to remove if that change appears to overload this PR

File: client/src/containers/DownloadProgressContainer.jsx [hunk 2]
```diff
@@ -66,6 +69,44 @@ class DownloadProgressContainer extends React.PureComponent {
...
+            <button
+              className="usa-button-outline"
+              onClick={this.restartDocumentDownload}
+              {...css({ marginLeft: '2rem' })}
+            >
+              Retry download
+            </button>
+          </li>
+        </ul>
+      </AlertBanner>;
```
Comment (lowellrex) [inline client/src/containers/DownloadProgressContainer.jsx]: We can return this AlertBanner any of three places in this function which seems a little fishy to me. However, I'm struggling for a way to factor them out without harming the readability of the code as it is now. Open to suggestions.

Comment (mdbenjam) [reply client/src/containers/DownloadProgressContainer.jsx]: I think it's fine.

File: client/src/containers/DownloadProgressContainer.jsx [hunk 3]
```diff
@@ -66,6 +69,44 @@ class DownloadProgressContainer extends React.PureComponent {
   restartDocumentDownload = () => this.props.startDocumentDownload(this.props.manifestId, this.props.csrfToken);
 
   completeBanner() {
+    const sourcesForStatus = {};
+
+    for (const status of Object.values(MANIFEST_SOURCE_FETCH_STATE)) {
```
Comment (mdbenjam) [inline client/src/containers/DownloadProgressContainer.jsx]: This section seems overly complex to me. It seems doing two filters to get the SUCCEEDED and FAILED would be sufficient here. And I think it'd improve readability.

**Pull request merged**